core/src/OpenMP/Kokkos_OpenMP_Instance.cpp

```diff
@@ -34,18 +34,8 @@
 namespace Kokkos {
 namespace Impl {
 
-void OpenMPInternal::acquire_lock() {
-  while (1 == desul::atomic_compare_exchange(&m_pool_mutex, 0, 1,
-                                             desul::MemoryOrderAcquire(),
-                                             desul::MemoryScopeDevice())) {
-    // do nothing
-  }
-}
-
-void OpenMPInternal::release_lock() {
-  desul::atomic_store(&m_pool_mutex, 0, desul::MemoryOrderRelease(),
-                      desul::MemoryScopeDevice());
-}
```

Comment:
After a quick look, it seems that `m_pool_mutex` is not used anymore and we can remove it.